redis/cluster.py

```diff
@@ -0,0 +1,1249 @@
+        'CLUSTER COUNT-FAILURE-REPORTS': int,
+        'CLUSTER COUNTKEYSINSLOT': int,
+        'CLUSTER DELSLOTS': bool,
+        'CLUSTER FAILOVER': bool,
+        'CLUSTER FORGET': bool,
+        'CLUSTER GETKEYSINSLOT': list,
+        'CLUSTER INFO': parse_info,
+        'CLUSTER KEYSLOT': int,
+        'CLUSTER MEET': bool,
+        'CLUSTER NODES': parse_cluster_nodes,
+        'CLUSTER REPLICATE': bool,
+        'CLUSTER RESET': bool,
+        'CLUSTER SAVECONFIG': bool,
+        'CLUSTER SET-CONFIG-EPOCH': bool,
+        'CLUSTER SETSLOT': bool,
+        'CLUSTER SLAVES': parse_cluster_nodes,
+        'CLUSTER SLOTS': parse_cluster_slots,
+        'ASKING': bool,
+        'READONLY': bool,
+        'READWRITE': bool,
+    }
+
+    RESULT_CALLBACKS = dict_merge(
+        list_keys_to_dict([
+            "PUBSUB CHANNELS",
+        ], parse_pubsub_channels),
+        list_keys_to_dict([
+            "PUBSUB NUMSUB",
+        ], parse_pubsub_numsub),
+        list_keys_to_dict([
+            "PUBSUB NUMPAT",
+        ], parse_pubsub_numpat),
+        list_keys_to_dict([
+            "KEYS",
+        ], merge_result),
+    )
+
+    def __init__(
+            self,
+            host=None,
+            port=6379,
+            startup_nodes=None,
+            cluster_error_retry_attempts=3,
+            skip_full_coverage_check=False,
+            reinitialize_steps=10,
+            read_from_replicas=False,
+            url=None,
+            **kwargs
+    ):
+        """
+        :startup_nodes:
+            List of nodes from which initial bootstrapping can be done
+        :host:
+            Can be used to point to a startup node
+        :port:
+            Can be used to point to a startup node
+        :skip_full_coverage_check:
+            Skips the check of cluster-require-full-coverage config, useful for
+            clusters without the CONFIG command (like aws)
+       :read_from_replicas:
+            Enable read from replicas in READONLY mode. You can read possibly
+            stale data.
+            When set to true, read commands will be assigned between the
+            primary and its replications in a Round-Robin manner.
+        :cluster_error_retry_attempts:
+
+        :**kwargs:
+            Extra arguments that will be sent into Redis instance when created
+            (See Official redis-py doc for supported kwargs
+        [https://github.com/andymccurdy/redis-py/blob/master/redis/client.py])
+            Some kwargs are not supported and will raise a
+            RedisClusterException:
+                - db (Redis do not support database SELECT in cluster mode)
+        """
+        if "db" in kwargs:
+            # Argument 'db' is not possible to use in cluster mode
+            raise RedisClusterException(
+                "Argument 'db' is not possible to use in cluster mode"
+            )
+
+        # Get the startup node/s
+        from_url = False
+        if not startup_nodes:
```

Comment:
PEP suggests we should actively None check like below. However, given your use case, maybe startup_nodes should default to an empty list. WDYT?

Reply:
changed